server.go

```diff
@@ -177,3 +181,19 @@ func (s *Server) DeserializeKE2(ke2 []byte) (*message.KE2, error) {
 func (s *Server) DeserializeKE3(ke3 []byte) (*message.KE3, error) {
 	return s.Parameters.DeserializeKE3(ke3)
 }
+
+// DeserializeAKEState sets the interal state of the AKE server from the given
+// bytes
+func (s *Server) DeserializeAKEState(state []byte) error {
```

Comment:
Do you think we can name this `SetState `as well? I'd be more telling that the state is modified, and the caller doesn't actually care whether it's serialized or not :)

```suggestion
func (s *Server) SetAkeState(state []byte) error {
```